p2p/protocol/circuitv2/client/conn.go

```diff
@@ -143,3 +144,20 @@ func (c *Conn) untagHop() {
 		delete(c.client.hopCount, p)
 	}
 }
+
+type capableConnWithStat interface {
+	tpt.CapableConn
+	network.ConnStat
+}
+
+type capableConn struct {
+	capableConnWithStat
+}
+
+var transportName = ma.ProtocolWithCode(ma.P_CIRCUIT).Name
+
+func (c *capableConn) ConnState() network.ConnectionState {
```

Comment:
Should we make this a value receiver instead? That lets us avoid pointer chasing.

Reply:
Done.